modules/database/src/ioc/db/dbEvent.c

```diff
@@ -1041,11 +1050,18 @@ static void event_task (void *pParm)
         pendexit = evUser->pendexit;
 
         evUser->pflush_seq++;
+        if(ellCount(&evUser->waiters)) {
+            /* hold lock throughout to avoid race between event trigger and destroy */
+            ELLNODE *cur;
+            for(cur = ellFirst(&evUser->waiters); cur; cur = ellNext(cur)) {
+                event_waiter *w = CONTAINER(cur, event_waiter, node);
+                if(w)
+                    epicsEventMustTrigger(w->wake);
```

Comment:
In the body of the `for (…; cur; …)` loop we know that `cur` can't be NULL, so how could `w` (which I think should point to the same address as `cur`) ever be NULL? Shouldn't that `if(w)` be `if (w->wake)` instead?

Reply:
You are correct, it should.

Reply:
Fixed.